clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -447,10 +500,34 @@ public ConfigDef withClientSaslSupport() {
             String joined = Utils.join(undefinedConfigKeys, ",");
             throw new ConfigException("Some configurations in are referred in the dependents, but not defined: " + joined);
         }
-        // parse all known keys
+        // Handle all known ConfigKey objects.
         Map<String, Object> values = new HashMap<>();
-        for (ConfigKey key : configKeys.values())
-            values.put(key.name, parseValue(key, props.get(key.name), props.containsKey(key.name)));
+        for (ConfigKey key : configKeys.values()) {
```

Comment:
Does this mean that we process the same key multiple times if there are deprecated names (given https://github.com/apache/kafka/pull/2456/files#diff-0abcccc4b2e75cfa2ded051cf47076beR123)? Is that what we want?

Reply:
The intention is that there should be only one ConfigDef entry for each logical configuration key.  The deprecated names are stored in a collection of Strings inside that ConfigDef object.